pandas/core/common.py

```diff
@@ -909,16 +909,17 @@ def diff(arr, n, axis=0):
         analagoust to s-s.shift(n) """
 
     n = int(n)
-    na = np.nan
     dtype = arr.dtype
+    if issubclass(dtype.type, np.integer):
```

Comment:
I created integer functions in generated.pyx for diff_2d, so that integers stayed as integers in the Cython. Previously the code assumed that the result would be a float so it used nan, and would throw an error with the new functions. This just made sure that instead of nan, we were using tslib.iNaT if it was integer data. Since I may be removing those diff_2d integer functions, this may be a moot point, but if they stay in, I'll add some testing.